components/service/firefox-accounts/src/main/java/mozilla/components/service/fxa/Config.kt

```diff
@@ -47,10 +47,12 @@ data class DeviceConfig(
  * @property supportedEngines A set of supported sync engines, exposed via [GlobalSyncableStoreProvider].
  * @property syncPeriodInMinutes Optional, how frequently periodic sync should happen. If this is `null`,
  * periodic syncing will be disabled.
+ * @property tokenServerUrl Optional, override default tokenServerUrl to support self-hosted sync storage.
  */
 data class SyncConfig(
```

Comment:
@grigoryk - So it seems that the ServerConfig (which was the suggestion on where to overwrite this value in the original [issue](https://github.com/mozilla-mobile/android-components/issues/3729)) is pulled from [here](https://github.com/mozilla/application-services/blob/master/components/fxa-client/android/src/main/java/mozilla/appservices/fxaclient/Config.kt) in application services and doesn't currently support setting a custom token server url.

Since #3729 was filed specifically for this repo, I wanted to first confirm with you that you still think it makes sense for the ServerConfig to set this value. If so, I can close this PR and first open a PR over in application services to support setting a token server url. Then, once that merges, I can re-open this one and adjust to use that value instead.

Since that's a much larger scope of changes, I figured it'd make sense to confirm this approach with you first.